### DISCUSSION TITLE
Some confusion about readOnly mode

### DISCUSSION THREAD

**Discussion by Alecyrus** [ueberdosis/hocuspocus#302] [Q&A] [answered]
Imagine there is a collaborative document (document owner and a collaborator) and if the document owner allows the collaborator to read and write to the document at the beginning. The collaborator will be able to read and write to the document as we expected when he first access it and have it cached locally (via indexdbPersistence).
But when the collaborator leaves the Internet (where he can still read and write to the document locally) and makes a lot of changes to the document locally, and at the same time the document owner changes the collaborator's permissions to read-only, so when the collaborator reconnects to the Internet, he will only have readonly permissions according to the logic of the onAuthenticate hook, will all changes made to the document by the collaborator when he is offline be discarded?

**Comment by georeith** [comment, marked as answer]
@Alecyrus my understanding is no the changes will not be discarded just not accepted by the server, the user will just have changes the server doesn't that it loads from indexeddb. 

AFAIK HocusPocus does not do any replace operations whatsoever on the client. It only ever adds updates. 

You would need to implement something that manually deletes the indexeddb copy and replaces it with the server copy if authentication fails.

I do something similar to expire documents on the server over time (remove their history as they grow large) and invalidate the clients copy after.
